api/src/main/java/com/cloud/agent/api/storage/OVFHelper.java

```diff
@@ -119,8 +120,7 @@ protected OVFPropertyTO createOVFPropertyFromNode(Node node, int index, String c
         boolean password = StringUtils.isNotBlank(passStr) && passStr.equalsIgnoreCase("true");
         String label = ovfParser.getChildNodeValue(node, "Label");
         String description = ovfParser.getChildNodeValue(node, "Description");
-        logger.debug("Creating OVF property index " + index + (category == null ? "" : " for category " + category)
-                + " with key = " + key);
+        logger.debug("Creating OVF property index {} {} with key = {}", index, (category == null ? "" : " for category " + category), key);
```

Comment:
You could use lambda here

Reply:
should I use lambdas even when there is no big structure being read from memory or costly function being called?

Reply:
We could avoid the if else and the string concatenation when debug is not turned on. But on a second thought, these are indeed not very costly considering the scope of ACS, so we can keep as is.